src/H5FDsubfiling/H5FDsubfiling.c

```diff
@@ -564,6 +564,117 @@ H5Pget_fapl_subfiling(hid_t fapl_id, H5FD_subfiling_config_t *config_out)
     FUNC_LEAVE_API(ret_value)
 } /* end H5Pget_fapl_subfiling() */
 
+herr_t
+H5FDsubfiling_get_file_mapping(hid_t file_id, char ***filenames, size_t *len)
+{
+    subfiling_context_t *sf_context    = NULL;
+    H5FD_t              *driver        = NULL;
+    H5F_t               *file_ptr      = NULL;
+    char               **filenames_arr = NULL;
+    char                *filepath      = NULL;
+    char                *subfile_dir   = NULL;
+    char                *base          = NULL;
+    int                  num_subfiles  = 0;
+    int                  num_digits    = 0;
+    herr_t               ret_value     = SUCCEED;
+
+    FUNC_ENTER_API(FAIL)
+
+    if (file_id < 0)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid file ID");
+    if (!filenames)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "`filenames` was NULL");
+    if (!len)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "`len` was NULL");
+
+    *filenames = NULL;
+    *len       = 0;
+
+    if (NULL == (file_ptr = H5VL_object(file_id)))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid file ID");
+
+    if (H5FD_SUBFILING != H5F_get_driver_id(file_ptr))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "file is not using Subfiling VFD");
+
+    if (H5F_shared_get_file_driver(H5F_SHARED(file_ptr), &driver) < 0)
+        HGOTO_ERROR(H5E_FILE, H5E_CANTGET, FAIL, "can't get driver structure from file ID");
+
+    if (NULL == (sf_context = H5FD__subfiling_get_object(((H5FD_subfiling_t *)driver)->context_id)))
+        HGOTO_ERROR(H5E_FILE, H5E_CANTGET, FAIL, "can't get subfiling context from ID");
+
+    if (!sf_context->topology)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL,
+                    "application topology hasn't been initialized yet for this file");
+
+    assert(sf_context->h5_file_id != UINT64_MAX);
+    assert(sf_context->h5_filename);
+    assert(sf_context->sf_num_subfiles > 0);
+    assert(sf_context->topology);
+
+    if (sf_context->topology->rank_is_ioc) {
+        assert(sf_context->sf_fids);
+        assert(sf_context->sf_num_fids > 0);
+
+        /* Get the basename of the full HDF5 filename */
+        if (H5_basename(sf_context->h5_filename, &base) < 0)
+            HGOTO_ERROR(H5E_RESOURCE, H5E_CANTALLOC, FAIL, "can't get HDF5 file basename");
```

Comment:
Was H5E_RESOURCE an explicit choice or from copy-paste? Major error codes are a mess in the library, but I'd probably use H5E_VFL instead of H5E_RESOURCE. It'd be nice to just standardize on the major error code matching the package.

Reply:
I've updated all the H5E_RESOURCE occurrences to use H5E_VFL instead.